src/translate.py

```diff
@@ -1299,13 +1300,22 @@ def should_write(self) -> bool:
         return True
 
     def format(self, fmt: Formatter) -> str:
-        source = self.source
+        dest = late_unwrap(self.dest)
+        source = late_unwrap(self.source)
         if (
-            isinstance(self.dest, StructAccess) and self.dest.late_has_known_type()
-        ) or isinstance(self.dest, (ArrayAccess, LocalVar, RegisterVar, SubroutineArg)):
+            isinstance(dest, StructAccess) and dest.late_has_known_type()
+        ) or isinstance(dest, (ArrayAccess, LocalVar, RegisterVar, SubroutineArg)):
             # Known destination; fine to elide some casts.
             source = elide_casts_for_store(source)
-        return f"{self.dest.format(fmt)} = {format_expr(source, fmt)};"
+        if isinstance(source, BinaryOp) and source.op in COMPOUND_ASSIGNMENT_OPS:
+            rhs = None
+            if late_unwrap(source.left) == dest:
```

Comment:
could also check `source.left.format() == dest.format()` if this turns out to be too strong